implementations/rust/ockam/ockam_transport_tcp/tests/send_receive.rs

```diff
@@ -93,3 +95,51 @@ async fn tcp_lifecycle__reconnect__should_not_error(ctx: &mut Context) -> Result
 
     Ok(())
 }
+
+#[ignore]
+#[ockam_macros::test(timeout = 400000)]
+async fn tcp_keepalive_test(ctx: &mut Context) -> Result<()> {
+    TcpTransport::create(ctx).await?;
+
+    let message: String = rand::thread_rng()
+        .sample_iter(&rand::distributions::Alphanumeric)
+        .take(256)
+        .map(char::from)
+        .collect();
+
+    // Send the message to the cloud node echoer
+    ctx.send(
+        route![(TCP, "1.node.ockam.network:4000"), "echo"],
+        message.to_string(),
+    )
+    .await?;
+
+    // Wait to receive an echo and print it.
+    let reply = ctx.receive::<String>().await?;
+    info!("Sender has received the following echo: {}\n", reply);
+
+    // Sleep the thread to allow the tcp socket to send keepalive probes
+    let sleep_duration = Duration::from_secs(350);
+    info!("Slepping thread now for {:?}", sleep_duration);
+    thread::sleep(sleep_duration);
```

Comment:
I think we don't need to hand the thread. How about replacing that with `ctx.sleep()` to use more async-friendly sleep?